containers/unit-tests/start

```diff
@@ -22,6 +23,34 @@ if test -z "${docker:=$(which podman || which docker || true)}"; then
   exit 1
 fi
 
-set -ex
-exec ${docker} run --shm-size=512M --volume "${top_srcdir}":/source:ro \
-       ${ccarg:+"$ccarg"} $flags -- "$image" $cmd
+# restore image from Semaphore cache
+if [ -d "${SEMAPHORE_CACHE_DIR:-}" ]; then
+    image_esc="${image//\//_}"
+    image_esc="${image_esc//:/_}"
+    for f in $SEMAPHORE_CACHE_DIR/docker/$image_esc-*; do
+        [ -e "$f" ] || continue
+        echo "Restoring cached Docker image $f..."
+        ${docker} load < "$f"
+    done
+fi
+
+set -x
+RC=0
```

Comment:
I don't understand what's going on here.  This trick with conditionally setting RC would be useful if we were keeping `set -e`, but you get rid of it here.  In that case, why not just unconditionally capture RC=$? after?

Reply:
This does set -x, not set -e :-) I initially had it wrapped into set +e/set -e, but I find this a bit easier.